lib/project_types/script/layers/application/supported_languages.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Application
+      class SupportedLanguages
+        STABLE_LANGUAGES = %w(AssemblyScript)
+        BETA_LANGUAGES = %w(Rust)
+        private_constant :STABLE_LANGUAGES, :BETA_LANGUAGES
+
+        def self.all
+          languages = STABLE_LANGUAGES
+          languages += BETA_LANGUAGES if ShopifyCli::Feature.enabled?(:scripts_beta_languages)
```

Comment:
How does the feature flag mechanism work? Can an external dev randomly enable a feature? If not, how do we prevent that? (This question has nothing to do with your change, it's mostly curiosity)

Reply:
Yeah, anyone can enable it if they know what the feature flag is. There really isn't anyway to prevent someone from enabling a feature since this is open source and anyone could edit the code anyway, so I think obscurity is fine here